design/ceph/object/ceph-rgw-k8s-sa-authentication.md

```diff
@@ -0,0 +1,195 @@
+---
+Service authentication with vault for RGW
+target-version: release-1.14
+---
+
+# Feature Name
+
+Service account authentication with vault for RGW.
+
+## Summary
+
+When `Vault` is set up in Kubernetes environment, it can be authenticated using service account token. The OSD encryption already supports this feature and adding same for RGW. It uses same set of  config option in `ConnectionDetails` for `KeyManagementServiceSpec`.
+
+### Goals
+
+Configure RGW with service account authentication with Vault with help of vault agent.
+
+### Non-Goals
+
+There are different ways RGW can authenticate with help of [vault agent](https://docs.ceph.com/en/latest/radosgw/vault/#vault-agent), but only service account authentication is supported here.
+
+## Proposal details
+
+The [vault agent](https://developer.hashicorp.com/vault/docs/agent-and-proxy/agent#vault-stanza) can be configure with help of following annotation in `CephObjectStore CRD` :
+
+- vault.hashicorp.com/agent-inject --> brings vault agent with rgw pod
```

Comment:
Rook can generate these annotations on the rgw pod if needed, but why require them as annotations on the CRD? Rook always expects settings on the CRD, rather than annotations for settings.

Reply:
So will use settings in the CRD and add the annotation on RGW pod based on settings. Sounds good to me